### PR TITLE
Adds a "Setting up a Rails project" section.

### PR DESCRIPTION
**Pull request by mejackreed**
Outlines current practice in DLSS on how many Rails projects are setup. The intent
here is for documenting for others how we work on Rails projects.

Hopefully not controversial, but providing a place to where we have this codified.

### CODE REVIEW

**Comment by darrenleeweber** [comment]
:+1: and some guidelines on `database.yml` might help too.

File: best-practices/setting_up_rails_projects.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Setting up Rails Projects
+
+Many of the DLSS software projects are [Ruby on Rails](http://rubyonrails.org/) applications. Consistency across projects can aid developers and reduce the overall maintenance burden on teams. This guide should be used as an outline on common expectations in DLSS Ruby on Rails projects. Not only should our projects reflect consistency internally, but should also reflect larger community expectations. These are not to the letter rules, but guidelines aimed to help the team.
```
Comment (mjgiarlo) [inline best-practices/setting_up_rails_projects.md]: Drop "to the letter?" It parses weird to me, and I don't think it adds much to your (very good) point.

File: best-practices/setting_up_rails_projects.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Open source be default
+DLSS projects should be open source by default. See [licensing](/licensing.md). Developers should _never_ check in api keys or private information. Use the [config](https://github.com/railsconfig/config) gem instead. If you have any questions, check with dev-ops first.
```
Comment (mjgiarlo) [inline best-practices/setting_up_rails_projects.md]: Not remembering how GitHub markdown handled relative links... do you want `/licensing.md` there or `/best-practices/licensing,md`?

Comment (mjgiarlo) [inline best-practices/setting_up_rails_projects.md]: Yup, agree it's out of scope. `#wontfix`, nm!

**Comment by mejackreed** [comment]
Updated to address @mjgiarlo and @darrenleeweber comments.

**Comment by mejackreed** [comment]
Thanks for the feedback all!

**Comment by mjgiarlo** [comment]
:+1:

**Comment by darrenleeweber** [comment]
In sul-pub we also had concerns about database limits in the `db/schema.rb` file.  Is there a guideline to apply for this also?

**Comment by mejackreed** [comment]
Thanks @darrenleeweber added a sentence on that.

**Comment by darrenleeweber** [comment]
:+1: for addressing all my comments.  (Leaving it open for others to review.  Do we have a designated merger for these playbook PRs?)

**Comment by mejackreed** [comment]
I think just get enough :+1: and ship it /shrug?

**Comment by azaroth42** [comment]
:ok_hand:  (non-binding approval)

File: best-practices/setting_up_rails_projects.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Setting up Rails Projects
+
+Many of the DLSS software projects are [Ruby on Rails](http://rubyonrails.org/) applications. Consistency across projects can aid developers and reduce the overall maintenance burden on teams. This guide should be used as an outline on common expectations in DLSS Ruby on Rails projects. Not only should our projects reflect consistency internally, but should also reflect larger community expectations. These are not rules, but guidelines aimed to help the team.
+
+## Open source be default
```
Comment (jmartin-sul) [inline best-practices/setting_up_rails_projects.md]: possible typo?  "Open source by default"?

Comment (mejackreed) [inline best-practices/setting_up_rails_projects.md]: Nice catch! most definitely a type.. pushed up a fix.

Comment (mjgiarlo) [inline best-practices/setting_up_rails_projects.md]: I thought he was affecting a pirate accent.

Comment (mejackreed) [inline best-practices/setting_up_rails_projects.md]: "All documentation must now be in a pirate accent"  -@mjgiarlo

**Pull request merged**